### PR TITLE
Update ce.py

### PR DESCRIPTION
**Pull request by KyloRen1**
## Description
Implementation of Symmetric Cross Entropy 

## Related Issue

https://github.com/catalyst-team/catalyst/issues/479

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [Code of Conduct](https://github.com/catalyst-team/catalyst/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I have read the [Contributing](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md) guide.
- [ ] I have checked the code-style using `make check-style`.
- [x] I have written the docstring in Google format for all the methods and classes that I used.
- [ ] I have checked the docs using `make check-docs`.

### CODE REVIEW

**Comment by Scitator** [comment]
@KyloRen1 Could you please also update __init__.py, so this criterion would be visible from contrib?

**Comment by KyloRen1** [comment]
@Scitator hope it works now

File: catalyst/contrib/criterion/ce.py [hunk 1]
```diff
@@ -14,3 +14,23 @@ def forward(self, input, target):
         loss = -torch.sum(input * target)
         loss = loss / input.size()[0] if self.size_average else loss
         return loss
+    
+class SymmetricCrossEntropyLoss(nn.Module):
+	def __init__(self, alpha, beta, num_classes = 10):
```
Comment (Scitator) [inline catalyst/contrib/criterion/ce.py]: please, remove `num_classes = 10`, we need to specify them

File: catalyst/contrib/criterion/ce.py [hunk 2]
```diff
@@ -14,3 +14,23 @@ def forward(self, input, target):
...
+		super(SymmetricCrossEntropyLoss, self).__init__()
+		self.alpha = alpha
+		self.beta = beta
+		self.num_classes = num_classes
+		self.device = 'cuda' if torch.cuda.is_available() else 'cpu'
```
Comment (Scitator) [inline catalyst/contrib/criterion/ce.py]: why do we need device here?
it's also a good way to use `utils.get_device()`

**Comment by Scitator** [comment]
@KyloRen1 please, check codestyle travis output :)

**Comment by KyloRen1** [comment]
@Scitator I have changed code and tried to format it correctly in GitHub UI. But, probably, due to lack of experience, I am still having troubles, as the Travis outputs errors. What could you recommend, thanks!

**Comment by Scitator** [comment]
@KyloRen1 just follow the travis output and try not to use GitHub UI – it's much easier to check the codestyle locally by following [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md).

**Comment by Scitator** [comment]
@KyloRen1 Hi, how is your contribution going? Is there any help needed?

**Comment by KyloRen1** [comment]
@Scitator Thanks for the proposal, unfortunately, I am a bit busy right now. I will try to finish this weekend

**Comment by KyloRen1** [comment]
@Scitator Hi! Seems like I finally figured out what was the issues.)

**Comment by Scitator** [comment]
Dear @KyloRen1 could you please write some docs for your criterion, so other community could easily understand it?

File: catalyst/contrib/criterion/ce.py [hunk 3]
```diff
@@ -14,3 +14,31 @@ def forward(self, input, target):
...
         loss = loss / input.size()[0] if self.size_average else loss
         return loss
+
+
+class SymmetricCrossEntropyLoss(nn.Module):
+    """
+    Symmetric Cross Entropy
+    paper : https://arxiv.org/abs/1908.06112
+    """
+    def __init__(self, alpha, beta, num_classes):
```
Comment (Scitator) [inline catalyst/contrib/criterion/ce.py]: could you please write docs for `alpha`, `beta`, `num_classes`? what do they mean?

Comment (Scitator) [reply catalyst/contrib/criterion/ce.py]: and their type?

File: catalyst/contrib/criterion/ce.py [hunk 4]
```diff
@@ -14,3 +14,31 @@ def forward(self, input, target):
...
+        super(SymmetricCrossEntropyLoss, self).__init__()
+        self.alpha = alpha
+        self.beta = beta
+        self.num_classes = num_classes
+
+    def forward(self, input, target):
```
Comment (Scitator) [inline catalyst/contrib/criterion/ce.py:35]: and the same thing here...
it would be also nice, if you could write the dimensions of the tensors, like
`input - [batch_size; num_features]` or something like that

**Comment by lightforever** [comment]
@KyloRen1, thank you for the collaboration!

Could you, please, add a test for the feature? The example:

Firstly, we need to add a config like that below. The file name should be something like symmetric_cross_entropy.config. In that file, we need to replace CrossEntropyLoss with your class.

https://github.com/catalyst-team/catalyst/blob/master/examples/_tests_mnist_stages/config5.yml

Secondly, there is a requirement to add an invocation of the test to that file:

https://github.com/catalyst-team/catalyst/blob/8c7312143c88f25e1aa1a75540d7457f0e30005e/bin/tests/check_dl.sh#L91

Thank you!

**Comment by Scitator** [comment]
@lightforever could your please uncheck [WIP] label when the PR would be ready?

**Comment by lightforever** [comment]
@Scitator Ok

**Comment by KyloRen1** [comment]
@lightforever Hi! Thanks for your comment! Did I understand correctly, that 2 point of your comment should look something like that
`PYTHONPATH=./examples:./catalyst:${PYTHONPATH} \
  python catalyst/dl/scripts/run.py \
  --expdir=./examples/_tests_mnist_stages \
  --config=./examples/_tests_mnist_stages/config5.yml \
  --logdir=./examples/logs/_tests_mnist_stages1 \
  --check
rm -rf ./examples/logs/_tests_mnist_stages1`

Thanks!

**Comment by lightforever** [comment]
@KyloRen1 , Hi! Yes, something like this with your config.

**Comment by Scitator** [comment]
hi guys, how is PR going? any help needed?

**Comment by KyloRen1** [comment]
Sorry for being late. I have had a few exams and just came home after the last one. Will fix the issues as soon as possible

**Comment by KyloRen1** [comment]
@lightforever Hi! It seems like I have fixed all the issues with my code. But I don't really get how can I fix the problems that Travis outputs. Could you please recommend? Thanks!
